### PR TITLE
fix: During diff creation if value changes from undefined to string evaluation is getting skipped

### LINKED ISSUE #21708: [Bug]: During diff creation if value changes from undefined to string evaluation is getting skipped
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When we change the table style to SHORT from DEFAULT, during diff as the value changed from undefined to SHORT translated diff converts it to NOOP and skips all dependent paths evaluation. Change here indicates if any path changes from no value to any value evaluate path and all dependent paths as well

### Steps To Reproduce

1.  create an app and add a Text widget - add value to {{Table1.pageSize}}
2. Drag and drop the table
3. go to style and change it to SHORT 
this is flaky so takes time to reproduce this problem 

### Public Sample App

_No response_

### Environment

Release

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by ApekshaBhosale**
## Description

When we change the table style to SHORT from DEFAULT, during diff as the value changed from undefined to SHORT translated diff converts it to NOOP and skips all dependent paths evaluation. Change here indicates if any path changes from no value to any value evaluate path and all dependent paths as well

- **MetaHOC changes**:- metaProperties is used to set the `initialMetaState` for the metaHOC instead of using `this.props`.

Fixes #21708 

## Type of change

> Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)

- Manual
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/ce/workers/Evaluation/evaluationUtils.ts [hunk 1]
```diff
@@ -210,17 +210,11 @@ export const translateDiffEventToDataTreeDiffEvent = (
       } else if (difference.lhs === undefined || difference.rhs === undefined) {
         // Handle static value changes that change structure that can lead to
         // old bindings being eligible
-        if (
-          difference.lhs === undefined &&
-          (isTrueObject(difference.rhs) || Array.isArray(difference.rhs))
-        ) {
+        if (difference.lhs === undefined && !!difference.rhs) {
```
Comment (ohansFavour) [inline app/client/src/ce/workers/Evaluation/evaluationUtils.ts:213]: if the rhs changes to a valid falsy (like an empty string, 0, false or null), this condition will not pass.

File: app/client/src/ce/workers/Evaluation/evaluationUtils.ts [hunk 2]
```diff
@@ -210,17 +210,11 @@ export const translateDiffEventToDataTreeDiffEvent = (
...
           result.event = DataTreeDiffEvent.NEW;
           result.payload = { propertyPath };
         }
-        if (
-          difference.rhs === undefined &&
-          (isTrueObject(difference.lhs) || Array.isArray(difference.lhs))
-        ) {
+        if (difference.rhs === undefined && !!difference.lhs) {
```
Comment (ohansFavour) [inline app/client/src/ce/workers/Evaluation/evaluationUtils.ts:217]: Same concern as above

File: app/client/src/workers/common/DataTreeEvaluator/index.ts [hunk 3]
```diff
@@ -381,8 +381,7 @@ export default class DataTreeEvaluator {
   } {
     const totalUpdateTreeSetupStartTime = performance.now();
 
-    let localUnEvalTree = Object.assign({}, unEvalTree);
-    // const localConfigTree = configTree;
+    let localUnEvalTree = klona(unEvalTree);
```
Comment (ohansFavour) [inline app/client/src/workers/common/DataTreeEvaluator/index.ts:384]: Is there a need to deep-clone here?. AFAIK, shallow copy works fine and takes less time to complete.

Comment (ApekshaBhosale) [reply app/client/src/workers/common/DataTreeEvaluator/index.ts:384]: i will remove this

File: app/client/src/ce/workers/Evaluation/evaluationUtils.ts (hunk 2 above)
Comment (ApekshaBhosale) [reply app/client/src/ce/workers/Evaluation/evaluationUtils.ts:217]: same as above

File: app/client/src/ce/workers/Evaluation/evaluationUtils.ts (hunk 1 above)
Comment (ApekshaBhosale) [reply app/client/src/ce/workers/Evaluation/evaluationUtils.ts:213]: so should i a particular type ie string, number, or boolean?

Comment (ohansFavour) [reply app/client/src/ce/workers/Evaluation/evaluationUtils.ts:213]: Use [isNil](https://lodash.com/docs/4.17.15#isNil).

Comment (ohansFavour) [reply app/client/src/ce/workers/Evaluation/evaluationUtils.ts:213]: `if (difference.lhs === undefined && !isNil(difference.rhs))`

**Comment by ohansFavour** [comment]
I have created an alternate task to clean up the logic for translating diffs (#21715). I will pick this up.

cc: @Rishabh-Rathod @ApekshaBhosale

**Review by Rishabh-Rathod — APPROVED** [review]
Changes LGTM.

**Pull request merged**